server/controllers/recipe-controller.js

```diff
@@ -47,11 +47,75 @@ class RecipeController {
         preparationTime
       })
         .then(({ dataValues }) => {
-          successResponse(res, { recipe: dataValues }, 201);
+          return successResponse(res, { recipe: dataValues }, 201);
         })
         .catch(err => errorResponse(res, err, 400));
     } catch (error) {
-      validationErrorResponse(res, error.details);
+      return validationErrorResponse(res, error.details);
+    }
+  }
+
+  /**
+   * Update Recipe
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @param {Funtion} next
+   * @memberof RecipeController
+   * @returns {undefined}
+   */
+  static async updateRecipe(req, res) {
+    const { id } = req.user;
+    const { slug } = req.params;
+    const { body } = req;
+    const { title, ingrident, steps, cookingTime, preparationTime } = req.body;
+    try {
+      await validate(body, recipeUpdateSchema);
+      const recipe = await Recipe.findOne({ where: { slug, userId: id } });
```

Comment:
I think this should be two separate things - finding the recipe, and checking if the person making the request is the person that created the recipe (those are two different errors, `Not Found` and `Forbidden`)

Reply:
ok, I will separate them.

Reply:
completed